pkg/server/server.go

```diff
@@ -417,15 +417,13 @@ func writeKubeConfig(certs string, config *Config) error {
 }
 
 func setupDataDirAndChdir(config *config.Control) error {
-	var err error
-
```

Comment:
How does this change fix the issue?

Reply:
the extra folder "server" appended in the path was originally set [here](https://github.com/dereknola/k3s/blob/master/pkg/server/server.go#L424): 
with the output path: `{user_dir}/.rancher/k3s/server/agent/etc/rancher/node/password`

The changes made in this PR (along with code refactoring) are visible here:
https://github.com/k3s-io/k3s/blob/c11ecb7cdd899820744628bfa94666b6b730d5fc/pkg/server/server.go#L419-L422
with the output path: `{user_dir}/.rancher/k3s/agent/etc/rancher/node/password`
as described in the issue: https://github.com/k3s-io/k3s/issues/3636

but since I got  integration tests errors, I may be missing something.

Reply:
@AkihiroSuda Do you have any feedback on this?